starter-code/routes/auth.js

```diff
@@ -0,0 +1,239 @@
+auth.get('/login', (req, res, next) => {
+  if(req.session.currentUser) {
+    res.render('auth/welcome'
+    ,{ username: req.session.currentUser.username,
+        name: req.session.currentUser.name,
+        image: req.session.currentUser.image,
+         job: req.session.currentUser.job,
+       company: req.session.currentUser.company,
+       summary: req.session.currentUser.summary,
+       id: req.session.currentUser.id}
+  );} else {
+    res.render('auth/login');
+  }
+});
+
+// LOG IN PAGE, SUBMIT USER LOG IN CREDENTIALS
+auth.post('/login',(req, res, next) => {
+  let username = req.body.username;
+  let password = req.body.password;
+// check fields aren't empty
+  if(username === "" || password === ""){
+    res.render('auth/login', {
+      errorMessage: "You need to enter your username AND your password"
+    });
+    return;
+  }
+// check log in username exists in the db
+  User.findOne({"username": username},(err, user) =>{
+    if(err||!user){
+      res.render('auth/login', {
+        errorMessage: "Hmm...this user doesn't exist. Check your username and try again"
+      });
+      return;
+    }
+    // if the usernam exists, encrypts password and checks this against stored password
+    if(bcrypt.compareSync(password, user.password)){
+      // store the session
+      req.session.currentUser = user;
+
+      res.render('auth/welcome'
+      ,{ username: req.session.currentUser.username,
+       name: req.session.currentUser.name,
+       image: req.session.currentUser.image,
+        job: req.session.currentUser.job,
+      company: req.session.currentUser.company,
+      summary: req.session.currentUser.summary,
+      id: req.session.currentUser.id}
+    );
+    } else {
+      res.render('auth/login', {
+        errorMessage: "Oops, that password isn't correct. Try again"
+      });
+    }
+  });
+});
+
+// EDIT CONTENT: takes the user to a page they can update info on about themselves
+auth.get('/:id/edit', (req, res, next) => {
+  if(req.session.currentUser) {
+    const userID = req.session.currentUser._id;
+    //returns the ID as a string -- may be important later, if needs to be a number
+    User.findById(userID, (err, user)=>{
+      if(err){
+        return next(err);
+      }
+      // if they're logged in and click to edit, show the edit.ejs file info
+      res.render('auth/edit', {
+        name: user.name,
+        username: user.username,
+        image: user.image,
+         job: user.job,
+       company: user.company,
+       summary: user.summary,
+        id: user.id
+      });
+    });
+  } else {
+    res.render('auth/login', {
+      errorMessage: "Oops, something went wrong. Please log in and try again"
+    });
+  }
+});
+
+// NEED TO CREATE THE EDIT PAGE AGAIN. TRY WITHOUT THE TEXT FIELD AND WITHOUT THE IMAGE FIRST
+auth.post('/:id', (req, res, next) => { // why is this not post('/:id/edit')?
```

Comment:
It could be /edit, but it would be a bit redundant. There are generel conventions about route (or "endpoints") naming, a commin paradigm being REST API. So, for example, an API for posts on a social media app might look like:
GET /posts - return all posts
GET /posts/:id - return specific post
POST /posts - add a new post
POST /posts/:id - edit a specific post

The last one would actually be a PUT verb usually (PUT /posts/:id ), although we haven't mentioned this in class yet. 
You can read more about it in tutorials online, such as this: 
http://www.andrewhavens.com/posts/20/beginners-guide-to-creating-a-rest-api/

Reply:
Usually we would use
GET for getting info generally or specifically
POST for adding new items
PUT for editing or changing items
DELETE for deleting

But theoretically we could create a completely different architecture, and it would work